Add new test/merge process to project board doc (#3006)
bbc/simorgh: docs/Project-Board-Guide.md

<!-- prettier-ignore -->
| Column Name               | Issues/Pull Requests that belong in this column | When to move card to the next column | Additional context |
| ------------------------- | ------------------------------------------------| ------------------------------------ |------------------- |

> | To-do | Contains issues that have been newly created and added to the Simorgh project. | Once the issue has been refined by the corresponding labelled team, then it can be manually moved to "Ready for Dev". | n/a |

Reviewer: ```suggestion
| To-do | Contains issues that have been newly created and added to the Simorgh project. | Once the issue has been refined by the corresponding labelled pod (aka team), then it can be manually moved to "Ready for Dev". | n/a |
```

| Ready for Dev | Contains issues that have been refined. | Once you have selected an issue to work on, you must assign yourself to it and manually move the issue to the "Issues in progress".| n/a |
| Externally blocked | Contains issues/PRs that are blocked by an external factor. | Manually move the issue/PR when it is no longer blocked. | n/a |
| Issues in progress | Contains issues that are in working progress. | If the PR that uses a [keyword to link the issue](https://help.github.com/en/articles/closing-issues-using-keywords) has been merged, then the cards in this column will move automatically to "Done". | PRs should not use keywords unless they definitely complete the entire issue. |

> | PR in progress | Contains cards that have been automatically added here once a PR is assigned to the project "Simorgh". | It is your responsibility to move the card to "Code Review", when you wish for it to be reviewed by others. Teams prioritise code reviews over actual development. | PRs should be created on first commit, so that the team can see what you are working on. Draft PRs are preferable until the code is ready for review, when it can be converted to a proper PR. |

Reviewer: ```suggestion
| PR in progress | Contains cards that have been automatically added here once a PR is assigned to the project "Simorgh". | It is your responsibility to move the card to "Code Review", when you wish for it to be reviewed by others. Teams prioritise code reviews over actual development. | Draft PRs should be created on first commit, so that the team can see what you are working on. Once the code is ready for review, it can be converted to a proper PR. |
```

| UX, A11Y, External Review | Contains issues that require an accessibility swarm, UX or External review. | When the reviewing party approves the PR, the card will then have to be manually moved to "Ready for test" or "Ready for merge" depending on whether it needs manual testing. | [Read more about accessibility swarms](https://bbc-news.github.io/accessibility-news-and-you/accessibility-swarms) |
| Ready for test | Contains PRs that are ready for the test engineers to test. | Once approved by a test engineer, it is their responsibility to update the PR with an approval and move the card to "Ready for merge". | n/a |

> | Ready for merge | Contains tickets that have been code reviewed | It is the respnsibility of the author or code reviewer to check at this stage if the ticket needs manual testing by a tester. If not, you can proceed with the merge process. If your PR has no CI (continuous integration) errors and the merge button is not disabled, then you can merge your PR and it will automatically be moved to "Done". For cleanliness, you must also delete your branch after clicking merge. | n/a |

Reviewer: ```suggestion
| Ready for merge | Contains tickets that have been code reviewed | It is the responsibility of the author or code reviewer to check at this stage if the ticket needs manual testing by a tester. If not, the author or code reviewer should proceed with the merge process. If your PR has no CI (continuous integration) errors and the merge button is not disabled, then you can merge your PR and it will automatically be moved to "Done". For cleanliness, you must also delete your branch after clicking merge. | n/a |
```
